bc/channel/tasks.py

```diff
@@ -38,3 +44,68 @@ def enqueue_text_status_for_group(group: Group, text: str) -> None:
     """
     for channel in group.channels.all():
         enqueue_text_status_for_channel(channel, text)
+
+
+def refresh_threads_access_token(channel_pk):
+    """
+    Task to refresh the access token for a Threads channel.
+    """
+    try:
+        channel = Channel.objects.get(pk=channel_pk)
+    except Channel.DoesNotExist:
+        logger.warning(
+            f"Trying to refresh Threads access token for channel {channel_pk} but it no longer exists."
+        )
+        return
+
+    if channel.service != Channel.THREADS:
+        logger.warning(
+            f"Trying to refresh Threads access token for a {channel.get_service_display()} channel. Aborting."
+        )
+        return
+
+    refresh_access_token_url = "https://graph.threads.net/refresh_access_token"
+    params = {
+        "grant_type": "th_refresh_token",
+        "access_token": channel.access_token,
+    }
+    response = requests.get(refresh_access_token_url, params=params)
+
+    if response.status_code != 200:
+        logger.error(
+            f"Failed to refresh access token for Threads channel {channel}:"
+            f" {response.status_code} {response.text}"
+        )
+        return
+
+    data = response.json()
+    new_access_token = data.get("access_token")
+    expires_in = data.get("expires_in")  # In seconds
+
+    if new_access_token is None or expires_in is None:
+        logger.error(
+            f"Missing 'access_token' or 'expires_in' in refresh access token response for Threads channel {channel}: {data}\n"
+            f"If the issue persists, a new access token can be retrieved manually with the script again."
+        )
+        return
+
+    channel.access_token = new_access_token
+    channel.save()
+
+    # Schedule the next token refresh
+    delay_seconds = (
+        expires_in - 86400
+    )  # Subtract one day to avoid expiration before the task runs
+    queue.enqueue_in(
+        timedelta(seconds=delay_seconds if delay_seconds > 0 else expires_in),
+        refresh_threads_access_token,
+        channel_pk=channel.pk,
+        retry=Retry(
+            max=settings.RQ_MAX_NUMBER_OF_RETRIES,
+            interval=settings.RQ_RETRY_INTERVAL,
+        ),
+    )
```

Comment:
As discussed with @ERosendo, although enqueueing a job within the execution of another job isn't the most elegant solution, it's good enough for this use case since the first one will most definitely have finished executing way before the second begins.